### ISSUE TITLE
KTS IFC issue on demo server

### ISSUE THREAD

**Issue by dmomy** [usnistgov/ACVP#1184]
I am trying to download the vectors for demo session 181048.  I can’t seem to get vectors for 579303 (KTS-IFC Sp800-56Br2). I was wondering if you can look into this.  It has been over 24 hour that I have been trying to download. 

Below is a response I received from Jason Arnold at NIST.      The response I have received indicates that the associated data pattern hasn't been provided.   My capabilities file has "supportsNullAssociatedData": true  so I would think that not providing the associated data would be correct.  I will post the capabilities file that I'm using to generate vectors. 

I received this message from Jason Arnold @ NIST

So after some digging and conversation with the developers, the issue is that you are not providing a fixed info construction (associated data) but also not stating you support a blank one for this vector set. On our end, our validator couldn’t handle this particular case, so we’ve pushed up a change to the code base so that this type of error can be handled more accurately in the future, once we’ve fully tested it internally. So thanks for asking the question so that we could track this down.

```json
 {
    "algorithm": "KTS-IFC",
    "revision": "Sp800-56Br2",
    "iutId": "CAFECAFE",
    "modulo": [
        2048
    ],
    "scheme": {
        "KTS-OAEP-basic": {
            "l": 256,
            "kasRole": [
                "initiator",
                "responder"
            ],
            "ktsMethods": {
                "encoding": [
                    "concatenation"
                ],
                "hashAlgs": [
                    "SHA2-256"
                ],
                "supportsNullAssociatedData": true
            },
            "macMethods": {},
            "ktsMethod": {
                "hashAlgs": [
                    "SHA2-256"
                ],
                "encoding": [
                    "concatenation"
                ]
            }
        }
    },
    "function": [
        "keyPairGen",
        "partialVal"
    ],
    "keyGenerationMethods": [
        "rsakpg1-basic"
    ],
    "fixedPubExp": "010001"
}
```

**Comment by Kritner** [comment]
The `ktsMethods` should actually be `ktsMethod` - I see it is labeled incorrectly in the specification in one place which I'll correct shortly.  

The particular issue here is because we're not validating that a `associatedDataPattern` is provided in cases where `supportsNullAssociatedData` is false - which would be the case here, since the registered `supportsNullAssociatedData` is under the erroneous `ktsMethods` rather than `ktsMethod`.  There's a release in the works to get this check within our validation of the capabilities json basically.

In the meantime, registering a KTS capability under a new test session with `"supportsNullAssociatedData": true` nested under `ktsMethod` should work.

**Comment by dmomy** [comment]
Thank you Russ,
I've got the capabilities file corrected.  Thank you for your help on this.
Dennis

**Comment by Kritner** [comment]
This change has been released to prod in release [v1.1.0.18](https://github.com/usnistgov/ACVP-Server/releases/tag/v1.1.0.18).
